tests/integration_tests/functional/test_cmd_line_start.py

```diff
@@ -189,3 +190,51 @@ def test_config_with_default_limit(test_microvm_with_api, vm_config_file):
     response_err += "All previous unanswered requests will be dropped.\" }"
     _, stdout, _stderr = utils.run_cmd(cmd_err)
     assert stdout.encode("utf-8") == response_err.encode("utf-8")
+
+@pytest.mark.parametrize(
+    "metadata_file",
+    ["framework/metadata.json"]
+)
+def test_config_start_with_metadata(test_microvm_with_api, metadata_file):
+    """
+    Test if a microvm configured with metadata from file has the metadata available.
+
+    @type: functional
+    """
+    test_microvm = test_microvm_with_api
+
+    vm_metadata_path = os.path.join(test_microvm.path, os.path.basename(metadata_file))
+    shutil.copy(metadata_file, vm_metadata_path)
+    test_microvm.metadata_file = vm_metadata_path
+
+    test_microvm.spawn()
+
+    response = test_microvm.mmds.get()
+    assert test_microvm.api_session.is_status_ok(response.status_code)
+
+    with open(metadata_file) as json_file:
+        assert response.json() == json.load(json_file)
+
+@pytest.mark.parametrize(
+    "metadata_file",
+    ["framework/metadata_nonexisting.json"]
+)
+def test_config_start_with_metadata_missingfile(test_microvm_with_api, metadata_file):
+    """
+    Test if a microvm is attempted to be configured with metadata from a file 
+    that doesn't exist it should fail.
+
+    @type: functional
+    """
+    test_microvm = test_microvm_with_api
+
+    vm_metadata_path = os.path.join(test_microvm.path, os.path.basename(metadata_file))
+    test_microvm.metadata_file = vm_metadata_path
+
+    with pytest.raises(FileNotFoundError):
```

Comment:
as far as I can tell, this test validates that the test framework will throw an exception. We should instead be checking that Firecracker itself fails to start
We could also add more negative test cases here, like using invalid JSON

Reply:
On looking into this it was testing a file not found on the firecracker socket...and not the metadata file.  Thanks @alindima for the pointer here. I have changed these existing tests and also added 2 news tests for:
* Invalid json file
* Config file & mmds file used with `--no-api`